src/server/conn.rs

```diff
@@ -0,0 +1,180 @@
+use crate::{rt::tokio_executor::TokioExecutor, util::PrependAsyncRead};
+use http::{Request, Response};
+use http_body::Body;
+use hyper::{
+    body::Incoming,
+    server::conn::{http1, http2},
+    service::Service,
+};
+use std::{error::Error as StdError, io::Cursor, marker::Unpin};
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite};
+
+const H2_PREFACE: &[u8] = b"PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
+
+/// Http1 or Http2 connection.
+pub struct AutoConnection {
+    h1: http1::Builder,
+    h2: http2::Builder<TokioExecutor>,
+}
+
+impl AutoConnection {
+    /// Create a new AutoConn.
+    pub fn new() -> Self {
+        Self {
+            h1: http1::Builder::new(),
+            h2: http2::Builder::new(TokioExecutor::new()),
```

Comment:
Likely want to allow the user to provide an executor themselves.

Reply:
I managed to do it by making `serve_connection` require `Executor<BoxFuture<'static, ()>>` bounds and creating an executor that does the conversion. Now the question is should we somehow make `TokioExecutor` default? If yes, how?

Reply:
Now with RC3 out, you should be able to use the `hyper::rt::bounds` that you added, right?

Reply:
Yep.